avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -716,7 +719,7 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2023-07-02-p
         enabled: enableStorageProfileBlobCSIDriver
       }
       diskCSIDriver: {
-        enabled: enableStorageProfileDiskCSIDriver
+        enabled: costAnalysisEnabled == true ? true : enableStorageProfileDiskCSIDriver
```

Comment:
@PixelRobots Why is `costAnalysisEnabled` being used for `diskCSIDriver`? This is supposed to go under `metricsProfile` as per docs: https://learn.microsoft.com/en-us/azure/templates/microsoft.containerservice/managedclusters?pivots=deployment-language-bicep#managedclustermetricsprofile

Reply:
I have added it under metrics profiule. i forgot to copy and paste that over from my master. (i have a lot of changes I can add here if needed, like keda and more). 

I have also added the skuTier check on the storage as thios feature needs the disk driver to be enabled, but also it does not work on the free tier of AKS.

Reply:
Thanks. I will start dropping them in.

I have been working on a user defined type and module for Aks maintenance windows configuration but not sure it is 100% the best but will share that one too.